Modules/_ssl.c

```diff
@@ -2996,6 +2996,119 @@ static PyType_Spec PySSLSocket_spec = {
     .slots = PySSLSocket_slots,
 };
 
+#ifdef _MSC_VER
+static const unsigned char *
+_get_cert_bytes(const X509 *cert, int *length)
+{
+    unsigned char *cert_bytes;
+    int cert_bytes_length = i2d_X509(cert, NULL);
+    if (cert_bytes_length <= 0)
+    {
+        return NULL;
+    }
+
+    cert_bytes = PyMem_RawMalloc(cert_bytes_length);
+    if (cert_bytes == NULL)
+    {
+        return NULL;
+    }
+
+    if (i2d_X509(cert, &cert_bytes) <= 0)
+    {
+        return NULL;
+    }
+
+    *length = cert_bytes_length;
+
+    /* i2d_X509 moves the input pointer to the end of the data */
+    return cert_bytes - cert_bytes_length;
+}
+
+
+static int
+_verify_callback(int preverify_ok, X509_STORE_CTX *ctx)
+{
+    int ret = 0;
+    /* windows api calls below do not check for hostname mismatch */
+    if (preverify_ok || X509_STORE_CTX_get_error(ctx) == X509_V_ERR_HOSTNAME_MISMATCH)
+    {
+        return preverify_ok;
+    }
+
+    HCERTSTORE store = CertOpenStore(CERT_STORE_PROV_MEMORY, 0, 0, CERT_STORE_DEFER_CLOSE_UNTIL_LAST_FREE_FLAG, NULL);
```

Comment:
There's a lot of lines in this function that are too long (the guideline is 79 characters, per PEP-7). Could you break them up across multiple lines?

Reply:
I've fixed most violations except line 3063 which has a long constant name and is a couple characters too long, does it make more sense there to enforce the rule or to leave it?

Reply:
I think it's fine.